src/App.tsx

```diff
@@ -14,11 +14,70 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  selectedGood: string,
+};
+
+class App extends React.Component<{}, State> {
+  state = {
+    selectedGood: 'Jam',
+  };
+
+  render() {
+    return (
+      <div className="App">
+        <h1 className="App__title">{this.state.selectedGood.length > 1 ? `${this.state.selectedGood} is selected` : 'No goods selected'}</h1>
+        <button
+          type="button"
+          className={`App__clearButton button is-light ${this.state.selectedGood === '' ? 'App__clearButton--hide is-hidden' : ''}`}
+          onClick={() => {
+            this.setState({ selectedGood: '' });
+          }}
+        >
+          Clear
+        </button>
+        <ul className="App__list">
+          {goodsFromServer.map((product) => (
+            <div key={product}>
+              <li
+                className={`App__list-item ${this.state.selectedGood === product ? 'App__list-item--selected' : ''}`}
+              >
+                {product}
+              </li>
+              <button
+                type="button"
+                id={product}
+                key={product + 2}
+                className={`App__button button is-light ${this.state.selectedGood === product ? 'App__button--selected is-hidden' : ''}`}
+                onClick={(event) => {
+                  this.setState(
+                    {
+                      selectedGood: event.currentTarget.id,
+                    },
+                  );
+                }}
```

Comment:
You shouldn't use event object in this task
In simple js we don't know the name of the element, on which user clicked on
But in react we store this value in product variable and can use this variable in our function
```suggestion
                onClick={() => {
                  this.setState(
                    {
                      selectedGood: product,
                    },
                  );
                }}
```